test/e2e/auth/audit.go

```diff
@@ -61,45 +71,123 @@ var _ = SIGDescribe("Advanced Audit [Feature:Audit]", func() {
 		framework.ExpectNoError(err, "failed to create audit-secret")
 		_, err = f.ClientSet.Core().Secrets(f.Namespace.Name).Get(secret.Name, metav1.GetOptions{})
 		framework.ExpectNoError(err, "failed to get audit-secret")
+		_, err = f.ClientSet.Core().Secrets(f.Namespace.Name).List(metav1.ListOptions{})
+		framework.ExpectNoError(err, "failed to list secrets")
+		_, err = f.ClientSet.Core().Secrets(f.Namespace.Name).Watch(watchOptions)
+		framework.ExpectNoError(err, "failed to create watch for secrets")
 		err = f.ClientSet.Core().Secrets(f.Namespace.Name).Delete(secret.Name, &metav1.DeleteOptions{})
 		framework.ExpectNoError(err, "failed to delete audit-secret")
 
-		expectedEvents := []auditEvent{{
-			method:    "create",
-			namespace: namespace,
-			uri:       fmt.Sprintf("/api/v1/namespaces/%s/pods", namespace),
-			response:  "201",
-		}, {
-			method:    "delete",
-			namespace: namespace,
-			uri:       fmt.Sprintf("/api/v1/namespaces/%s/pods/%s", namespace, pod.Name),
-			response:  "200",
-		}, {
-			method:    "create",
-			namespace: namespace,
-			uri:       fmt.Sprintf("/api/v1/namespaces/%s/secrets", namespace),
-			response:  "201",
-		}, {
-			method:    "get",
-			namespace: namespace,
-			uri:       fmt.Sprintf("/api/v1/namespaces/%s/secrets/%s", namespace, secret.Name),
-			response:  "200",
-		}, {
-			method:    "delete",
-			namespace: namespace,
-			uri:       fmt.Sprintf("/api/v1/namespaces/%s/secrets/%s", namespace, secret.Name),
-			response:  "200",
-		}}
+		_, err = f.ClientSet.Core().ConfigMaps(f.Namespace.Name).Create(configMap)
+		framework.ExpectNoError(err, "failed to create audit-configmap")
+		_, err = f.ClientSet.Core().ConfigMaps(f.Namespace.Name).Get(configMap.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to get audit-configmap")
+		_, err = f.ClientSet.Core().ConfigMaps(f.Namespace.Name).List(metav1.ListOptions{})
+		framework.ExpectNoError(err, "failed to list config maps")
+		_, err = f.ClientSet.Core().ConfigMaps(f.Namespace.Name).Watch(watchOptions)
+		framework.ExpectNoError(err, "failed to create watch for config maps")
+		err = f.ClientSet.Core().ConfigMaps(f.Namespace.Name).Delete(configMap.Name, &metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "failed to delete audit-configmap")
+
+		config, err := framework.LoadConfig()
+		if err != nil {
+			framework.Failf("failed to load config: %v", err)
+		}
+		apiExtensionClient, err := clientset.NewForConfig(config)
+		if err != nil {
+			framework.Failf("failed to initialize apiExtensionClient: %v", err)
+		}
+		crd := testserver.NewRandomNameCustomResourceDefinition(apiextensionsv1beta1.ClusterScoped)
+		_, err = testserver.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.ClientPool)
+		testserver.DeleteCustomResourceDefinition(crd, apiExtensionClient)
+
+		expectedEvents := []auditEvent{}
+		expectedEvents = append(expectedEvents, commonExpectedEvents("pods", namespace, "create", pod.Name)...)
```

Comment:
better remove those appends and just create a `[]auditEvent{}` directly with all members. Should be much easier to read.

Reply:
In theory I can do it for commonExpectedEvents (this func returns only one event). But that would not work for  watchExpectedEvents and customResourceEvents. At least I haven't found in Go how to init slice using multiple other slices.

Reply:
oh, didn't see those `...`. There is no way in Go I fear.